Fix a bug in rgb2hsl method - thx Cedric for finding

From: Thorsten Behrens <tbehrens@novell.com>


---

 basegfx/source/color/bcolortools.cxx |    2 +-
 basegfx/test/basegfx2d.cxx           |    2 ++
 2 files changed, 3 insertions(+), 1 deletions(-)


diff --git basegfx/source/color/bcolortools.cxx basegfx/source/color/bcolortools.cxx
index 5e55e55..d20887d 100644
--- basegfx/source/color/bcolortools.cxx
+++ basegfx/source/color/bcolortools.cxx
@@ -65,7 +65,7 @@ namespace basegfx { namespace tools
             else if( g == maxVal )
                 h = 2.0 + (b - r)/d;
             else
-                h = 4.0 + (r - h)/d;
+                h = 4.0 + (r - g)/d;
 
             h *= 60.0;
 
diff --git basegfx/test/basegfx2d.cxx basegfx/test/basegfx2d.cxx
index 1bb14b8..3eb6d7e 100644
--- basegfx/test/basegfx2d.cxx
+++ basegfx/test/basegfx2d.cxx
@@ -1497,6 +1497,8 @@ public:
                                tools::rgb2hsl(maMagenta) == BColor(300,1,0.5));
         CPPUNIT_ASSERT_MESSAGE("cyan", 
                                tools::rgb2hsl(maCyan) == BColor(180,1,0.5));
+        CPPUNIT_ASSERT_MESSAGE("third hue case", 
+                               tools::rgb2hsl(BColor(0,0.5,1)) == BColor(210,1,0.5));
 
         CPPUNIT_ASSERT_MESSAGE("roundtrip white", 
                                tools::hsl2rgb(tools::rgb2hsl(maWhite)) == maWhite);
